File: fp1.scala
sealed trait Color {
  def isChromatic: Boolean = this match {
    case GrayScale => false
    case RGB => true
  }
}
case object GrayScale extends Color
case object RGB extends Color

object ColorUtil {
  def isChromatic(color: Color): Boolean = color match {
    case GrayScale => false
    case RGB => true
  }
}

println("ColorUtil.isChromatic(GrayScale) = " + ColorUtil.isChromatic(GrayScale)) 
println("RGB.isChromatic = " + RGB.isChromatic)

File: fp1.java
enum Color {
    GrayScale {
        boolean isChromatic() {
            return false;
        }
    },
    RGB {
        boolean isChromatic() {
            return true;
        }
    };

    abstract boolean isChromatic();
}

class ColorUtil {
    static boolean isChromatic(Color color) {
        switch (color) {
            case GrayScale: return false;
            case RGB: return true;
        }
        return false;
    }
}

System.out.println("ColorUtil.isChromatic(Color.GrayScale) = " + ColorUtil.isChromatic(Color.GrayScale));
System.out.println("Color.RGB.isChromatic() = " + Color.RGB.isChromatic());

/exit
